### PR TITLE
Moves helm chart templating to task interface

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/architect/issues/73

This changeset refactors the existing helm templating to a `TemplateHelmChartTask`, which satisfies the `Task` interface introduced earlier.

```
2017/07/11 10:52:22 running workflow: {
	docker-login:	'docker login --username=giantswarm+draughtsman --password=[REDACTED] quay.io'
	docker-tag-latest:	'docker tag quay.io/giantswarm/api:c55175c3d5cb24d67ff9cf20f8f2ed7687732885 quay.io/giantswarm/api:latest'
	docker-push-sha:	'docker push quay.io/giantswarm/api:c55175c3d5cb24d67ff9cf20f8f2ed7687732885'
	docker-push-latest:	'docker push quay.io/giantswarm/api:latest'
	template-helm-chart:	./helm c55175c3d5cb24d67ff9cf20f8f2ed7687732885
	helm-login:	'docker run --rm=false -v /Users/joseph/.cnr:/root/.cnr/ -w  quay.io/giantswarm/docker-helm:006b0db51ec484be8b1bd49990804784a9737ece registry login --user=giantswarm+draughtsman --password=[REDACTED] quay.io'
	helm-push:	'docker run --rm=false -v /Users/joseph/.cnr:/root/.cnr/ -v /Users/joseph/go/src/github.com/giantswarm/api/helm/api-chart:/Users/joseph/go/src/github.com/giantswarm/api/helm/api-chart -w /Users/joseph/go/src/github.com/giantswarm/api/helm/api-chart quay.io/giantswarm/docker-helm:006b0db51ec484be8b1bd49990804784a9737ece registry push --namespace=giantswarm --force quay.io'
}
```

So, the helm chart templating is now a task, and can be handled nicer in the future.
I'll refactor the kubectl stuff next.

### CODE REVIEW

File: template/helm.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	if err != nil {
+		return microerror.MaskAny(err)
+	}
+
+	if len(fileInfos) == 0 {
+		return nil
+	}
+
+	if len(fileInfos) > 1 {
+		return multipleHelmChartsError
```
Comment (xh3b4sd) [inline template/helm.go]: The error masking is missing here.

File: template/helm.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+	return nil
+}
+
+func (t TemplateHelmChartTask) Name() string {
+	return TemplateHelmChartTaskName
+}
+
+func (t TemplateHelmChartTask) String() string {
+	return fmt.Sprintf("%s:\t%s %s", t.Name(), t.helmPath, t.sha)
```
Comment (xh3b4sd) [inline template/helm.go]: Personal preference note. I would like to have such format strings as a constant with some useful godoc explaining the structure and maybe even providing some examples. That helps at least me personally to reason about the magic that is going on.

File: template/helm.go (hunk 1 above)
Comment (JosephSalisbury) [inline template/helm.go]: Good catch, fixed

File: template/helm.go (hunk 2 above)
Comment (JosephSalisbury) [inline template/helm.go]: Yeah, fixed. Brains

**Pull request merged**